resources/waterheater.rb

```diff
@@ -1215,8 +1255,6 @@ def self.create_new_heater(name, cap, fuel, vol, ef, re, jacket_r, t_set, space,
 
     new_heater.setHeaterMinimumCapacity(0.0)
     new_heater.setHeaterMaximumCapacity(UnitConversions.convert(cap, "kBtu/hr", "W"))
-    new_heater.setHeaterFuelType(HelperMethods.eplus_fuel_map(fuel))
-    new_heater.setHeaterThermalEfficiency(eta_c)
```

Comment:
I see what you're trying to do here, but it's a bit dangerous to only set the fuel type or thermal efficiency based on whether the UA is passed. That assumes that we'll never want to set the fuel/efficiency when UA is passed and will always want to set the fuel/efficiency when it's not passed. That is true of the current systems, but may not be true in the future. Rather, I would keep these setters here but only apply them if their value (`fuel` or `eta_c`) is not nil. Then someone could create a water heater in the future where they supply fuel, efficiency, and UA, for example.